modules/apps/analytics/analytics-settings-web/src/main/java/com/liferay/analytics/settings/web/internal/portlet/action/BaseAnalyticsMVCRenderCommand.java

```diff
@@ -89,6 +89,7 @@ protected String getJspPath() {
 	@Reference
 	protected ConfigurationProvider configurationProvider;
 
+	@Reference
```

Comment:
@JanisZhang This is not what I meant. You shouldn't add the fields to sub-classes; instead, just remove the setServletContext methods. For this reference, you need to keep the `target`, otherwise you don't know which servlet context will be injected.